extcap_ot.py

```diff
@@ -0,0 +1,205 @@
+
+    DirtylogFile = open(os.path.join(tempfile.gettempdir(), 'dirtylog'), 'w')
+    print('extcap {version=1.0.0}{display=OpenThread Sniffer}{help=https://github.com/openthread/pyspinel}')
+
+    for interface in comports():
+        th = threading.Thread(target=serialopen, args=(interface, __console__, DirtylogFile))
+        th.start()
+
+def extcap_capture(interface, fifo, control_in, control_out, channel, tap):
+    """Start the sniffer to capture packets"""
+    # baudrate = detect_baudrate(interface)
+    interface_port = str(interface).split(':')[0]
+    interface_baudrate = str(interface).split(':')[1]
+
+    if interface_baudrate is not None:
+        cmd = ['sniffer.py', '-c', channel, '-u', interface_port, '--rssi', '-b', interface_baudrate, '-o', str(fifo)]
+        print(cmd)
+
+        if tap:
+            cmd.append('--tap')
+        subprocess.Popen(cmd).wait()
+
+def extcap_close_fifo(fifo):
+    """"Close extcap fifo"""
+    # This is apparently needed to workaround an issue on Windows/macOS
+    # where the message cannot be read. (really?)
+    fh = open(fifo, 'wb', 0)
+    fh.close()
+
+
+if __name__ == '__main__':
+
+    # Capture options
+    parser = argparse.ArgumentParser(description='OpenThread Sniffer extcap plugin')
+
+    # Extcap Arguments
+    parser.add_argument('--extcap-interfaces', help='Provide a list of interfaces to capture from', action='store_true')
+    parser.add_argument('--extcap-interface', help='Provide the interface to capture from')
+    parser.add_argument('--extcap-dlts', help='Provide a list of dlts for the given interface', action='store_true')
+    parser.add_argument('--extcap-config', help='Provide a list of configurations for the given interface', action='store_true')
+    parser.add_argument('--extcap-reload-option', help='Reload elements for the given option')
+    parser.add_argument('--capture', help='Start the capture routine', action='store_true')
+    parser.add_argument('--fifo', help='Use together with capture to provide the fifo to dump data to')
+    parser.add_argument('--extcap-capture-filter', help='Used together with capture to provide a capture filter')
+    parser.add_argument('--extcap-control-in', help='Used to get control messages from toolbar')
+    parser.add_argument('--extcap-control-out', help='Used to send control messages to toolbar')
+    parser.add_argument('--extcap-baudrate', help='Auto detect the baudrate of the given interface', action='store_true')
+    parser.add_argument('--extcap-version', help='Wireshark Version')
+
+    # Interface Arguments
+    parser.add_argument('--channel', help='IEEE 802.15.4 capture channel [11-26]')
+    parser.add_argument('--tap', help='IEEE 802.15.4 TAP (only for Wireshark3.0 and later)', action='store_true')
+
+    try:
+        args, unknown = parser.parse_known_args()
+    except argparse.ArgumentError as e:
+        print('%s' % str(e))
+        fifo_found = False
+        fifo = ''
+        for arg in sys.argv:
```

Comment:
the code looks strange ? why is it necessary ?

Reply:
I refered to the example provided by Wireshark: [link](https://github.com/wireshark/wireshark/blob/efe1b0b1b2eab9312399128654b956d2f4472be5/doc/extcap_example.py#L452)
But I don't think it is necessary now, thank you.